pkg/generate/rbac/parser.go

```diff
@@ -0,0 +1,145 @@
+	var rbacRules []rbacv1.PolicyRule
+	fset := token.NewFileSet()
+
+	err := filepath.Walk(dir,
+		func(path string, info os.FileInfo, err error) error {
+			if !isGoFile(info) {
+				// TODO(droot): enable this output based on verbose flag
+				// fmt.Println("skipping non-go file", path)
+				return nil
+			}
+			rules, err := parseFile(fset, path, nil)
+			if err == nil {
+				rbacRules = append(rbacRules, rules...)
+			}
+			return err
+		})
+	return rbacRules, err
+}
+
+// filter function to ignore files from parsing.
+func isGoFile(f os.FileInfo) bool {
+	// ignore non-Go or Go test files
+	name := f.Name()
+	return !f.IsDir() &&
+		!strings.HasPrefix(name, ".") &&
+		!strings.HasSuffix(name, "_test.go") &&
+		strings.HasSuffix(name, ".go")
+}
+
+// parseFile parses given filename or content src and parses RBAC annotations
+// into RBAC rules.
+func parseFile(fset *token.FileSet, filename string, src interface{}) ([]rbacv1.PolicyRule, error) {
+	var rules []rbacv1.PolicyRule
+
+	f, err := parser.ParseFile(fset, filename, src, parser.ParseComments)
+	if err != nil {
+		return nil, err
+	}
+
+	// using commentMaps here because it sanitizes the comment text by removing
+	// comment markers, compresses newlines etc.
+	cmap := ast.NewCommentMap(fset, f, f.Comments)
+
+	for _, commentGroup := range cmap.Comments() {
+		for _, comment := range strings.Split(commentGroup.Text(), "\n") {
+			comment := strings.TrimSpace(comment)
+			if strings.HasPrefix(comment, "+rbac") {
+				if ann := getAnnotation(comment, "rbac", ":"); ann != "" {
+					rules = append(rules, parseRBACTag(ann))
+				}
+			}
+			if strings.HasPrefix(comment, "+kubebuilder:rbac") {
+				if ann := getAnnotation(comment, "kubebuilder:rbac", ":"); ann != "" {
+					rules = append(rules, parseRBACTag(ann))
+				}
+			}
+		}
+	}
+	return rules, nil
+}
+
+// getAnnotation returns the value of the comment with a prefix and separator.  E.g. for "name" and "="
+// "+name=foo" would return "foo"
+func getAnnotation(c, name, sep string) string {
+	prefix := fmt.Sprintf("+%s%s", name, sep)
+	if strings.HasPrefix(c, prefix) {
+		return strings.TrimPrefix(c, prefix)
+	}
+	return ""
+}
+
+// parseRBACTag parses the given RBAC annotation in to an RBAC PolicyRule.
+// This is copied from Kubebuilder code.
+func parseRBACTag(tag string) rbacv1.PolicyRule {
+	result := rbacv1.PolicyRule{}
+	for _, elem := range strings.Split(tag, ",") {
+		kv := strings.Split(elem, "=")
+		if len(kv) != 2 {
+			log.Fatalf("// +kubebuilder:rbac: tags must be key value pairs.  Expected "+
+				"keys [groups=<group1;group2>,resources=<resource1;resource2>,verbs=<verb1;verb2>] "+
+				"Got string: [%s]", tag)
+		}
+		value := kv[1]
+		if strings.HasPrefix(value, "\"") && strings.HasSuffix(value, "\"") {
+			value = value[1 : len(value)-1]
+		}
+		values := strings.Split(value, ";")
+		switch kv[0] {
+		case "groups":
+			normalized := []string{}
+			for _, v := range values {
+				if v == "core" {
```

Comment:
```
// +rbac:groups="",resources=pods,verbs=get;list;watch;delete
```
IMO using an empty string for core group should also be valid.
WDYT?

Reply:
Yes, it is. Current code is supporting both "" and "core".
Not dropping the support for "core" yet because old projects might have them, so it might help with the migration.